Amplify/Categories/API/ClientBehavior/APICategoryGraphQLBehavior.swift

```diff
@@ -117,4 +119,12 @@ public protocol APICategoryGraphQLBehavior: class {
                    listener: GraphQLSubscriptionOperation<AnyModel>.EventListener?)
     -> GraphQLSubscriptionOperation<AnyModel>
 
+    /// Attempts to create and start a reachability client for a host that corresponds to the apiName, and then
+    /// returns the associated Publisher which vends ReachabiltyUpdates
+    /// - Parameters:
+    ///   - apiName: The corresponding apiName that maps to the plugin configuration
```

Comment:
should apiName be optional? all other APIs operate on apiName being optional and then retrieving the single endpoint per endpointType (GraphQL|REST)

Reply:
What about something like `reachabilityPublisher(for apiName: String?)` which would default to "I will derive the API name if you have a single API configured".

We don't want to yet expose `allowsCellularAccess` until we get customer feedback on it. At that time we will need to plumb that decision/configuration through to the configuration of the NSURLSession for REST APIs and GraphQL queries/mutations, and to the WSS layer for GraphQL subscriptions, as well as reachability determinations. (Unless I'm misremembering and we already support this flag in the API category configuration)